pkg/prefix/prefix.go

```diff
@@ -0,0 +1,85 @@
+package prefix
+
+import (
+	"context"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/util/metautils"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type ctxKeyType string
```

Comment:
I noticed grpc uses an empty struct for their private context keys. Maybe we should adopt that over the string type.

Reply:
That's actually really smart.